clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -2041,9 +2041,11 @@ TemplateName TemplateInstantiator::TransformTemplateName(
         // We're rewriting the template parameter as a reference to another
         // template parameter.
         if (Arg.getKind() == TemplateArgument::Pack) {
-          assert(Arg.pack_size() == 1 && Arg.pack_begin()->isPackExpansion() &&
```

Comment:
nit: this code is duplicated 3 times, I think we can pull out a common function for it.